File: bindings/go/ldap.go
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>

// LDAP protocol types for proven-servers.
// All tag values match the Idris2 ABI discriminants exactly.
package proven

// SessionState represents the SessionState type (Idris2 ABI tags).
type SessionState uint8

const (
	SessionStateAnonymous SessionState = iota
	SessionStateBound
	SessionStateClosed
	SessionStateBinding
)

// Operation represents the Operation type (Idris2 ABI tags).
type Operation uint8

const (
	OperationBind Operation = iota
	OperationUnbind
	OperationSearch
	OperationModify
	OperationAdd
	OperationDelete
	OperationModDn
	OperationCompare
	OperationAbandon
	OperationExtended
)

// SearchScope represents the SearchScope type (Idris2 ABI tags).
type SearchScope uint8

const (
	SearchScopeBaseObject SearchScope = iota
	SearchScopeSingleLevel
	SearchScopeWholeSubtree
)

// ResultCode represents the ResultCode type (Idris2 ABI tags).
type ResultCode uint8

const (
	ResultCodeSuccess ResultCode = iota
	ResultCodeOperationsError
	ResultCodeProtocolError
	ResultCodeTimeLimitExceeded
	ResultCodeSizeLimitExceeded
	ResultCodeAuthMethodNotSupported
	ResultCodeNoSuchObject
	ResultCodeInvalidCredentials
	ResultCodeInsufficientAccessRights
	ResultCodeBusy
	ResultCodeUnavailable
)

File: bindings/ruby/lib/proven_servers/ldap.rb
# SPDX-License-Identifier: PMPL-1.0-or-later
# Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
#
# LDAP protocol types for proven-servers.

# frozen_string_literal: true

module ProvenServers
  # LDAP protocol types for proven-servers.
  module Ldap
    # SessionState matching the Idris2 ABI tags.
    module SessionState
      ANONYMOUS = 0
      BOUND = 1
      CLOSED = 2
      BINDING = 3
    end

    # Operation matching the Idris2 ABI tags.
    module Operation
      BIND = 0
      UNBIND = 1
      SEARCH = 2
      MODIFY = 3
      ADD = 4
      DELETE = 5
      MOD_DN = 6
      COMPARE = 7
      ABANDON = 8
      EXTENDED = 9
    end

    # SearchScope matching the Idris2 ABI tags.
    module SearchScope
      BASE_OBJECT = 0
      SINGLE_LEVEL = 1
      WHOLE_SUBTREE = 2
    end

    # ResultCode matching the Idris2 ABI tags.
    module ResultCode
      SUCCESS = 0
      OPERATIONS_ERROR = 1
      PROTOCOL_ERROR = 2
      TIME_LIMIT_EXCEEDED = 3
      SIZE_LIMIT_EXCEEDED = 4
      AUTH_METHOD_NOT_SUPPORTED = 5
      NO_SUCH_OBJECT = 6
      INVALID_CREDENTIALS = 7
      INSUFFICIENT_ACCESS_RIGHTS = 8
      BUSY = 9
      UNAVAILABLE = 10
    end

  end
end
